src/fifo.h

```diff
@@ -0,0 +1,41 @@
+/* A space/time efficient FIFO queue of pointers.
+ *
+ * Implemented with an unrolled single-linked list, the implementation packs multiple pointers into
+ * a single block.  This increases space efficiency and cache locality over the Redis `list` for the
+ * purpose of a simple FIFO queue.
+ */
+#ifndef __FIFO_H_
+#define __FIFO_H_
+
+typedef struct Fifo Fifo;
```

Comment:
All of the other Valkey types start with a lowercase letter.  Wondering if we should rename this to `fifo`?

At the time, I was thinking in terms of Java objects and focused on presenting an OO type structure.  Given that the capitalization seems anti-norm for Valkey, I think maybe we should change it.